presto-hdfs-core/src/main/java/com/facebook/presto/hive/BlockLocation.java

```diff
@@ -83,6 +86,13 @@ public long getLength()
         return length;
     }
 
+    public long getRetainedSizeInBytes()
+    {
+        int hostListSizeInBytes = hosts.stream().map(String::length).reduce(0, Integer::sum);
```

Comment:
is it possible that this value crosses int limit? If not it's fine else may be convert this into Long as we are doing it for `customSplitInfo` below.

Reply:
done. converted to Long.